lua/vgit/cli/Git.lua

```diff
@@ -7,21 +7,45 @@ local Hunk = require('vgit.cli.models.Hunk')
 local Log = require('vgit.cli.models.Log')
 local File = require('vgit.cli.models.File')
 local Blame = require('vgit.cli.models.Blame')
+local git_setting = require('vgit.settings.git')
 
 -- TODO: Expensive jobs can be run in the background by leveraging coroutines.
 -- The decision to make a job run in background cannot be controlled, which needs to changed.
 
+local function table_merge(table1, table2)
+  table1 = vim.deepcopy(table1)
```

Comment:
I think we can avoid the `deepcopy` here. As a general rule of thumb I try to avoid deep copies, unless I am dealing with nested objects.

Checkout [`utils.list.concat` ](https://github.com/tanvirtin/vgit.nvim/blob/main/lua/vgit/core/utils.lua#L242) you should be able to replace `table_merge` with it.

Reply:
Not possible ATM, without `vim.deepcopy` using `utils.list.concat(self.fallback_args, {...})` modifies `self.fallback_args` which obviously messes things up badly :-)

Reply:
Ah right, `utils.list.concat` is modifying the existing list. Okay this works!

Reply:
Ah right, `utils.list.concat` is modifying the existing list. Okay this works!